pySDC/helpers/pySDC_as_gusto_time_discretization.py

```diff
@@ -0,0 +1,188 @@
+import firedrake as fd
+
+from gusto.time_discretisation.time_discretisation import TimeDiscretisation, wrapper_apply
+from gusto.core.labels import explicit
+
+from pySDC.implementations.controller_classes.controller_nonMPI import controller_nonMPI
+from pySDC.implementations.problem_classes.GenericGusto import GenericGusto, GenericGustoImex
+from pySDC.core.hooks import Hooks
+from pySDC.helpers.stats_helper import get_sorted
+
+
+class LogTime(Hooks):
+    """
+    Utility hook for knowing how far we got when using adaptive step size selection.
+    """
+
+    def post_step(self, step, level_number):
+        L = step.levels[level_number]
+        self.add_to_stats(
+            process=step.status.slot,
+            process_sweeper=L.sweep.rank,
+            time=L.time,
+            level=-1,
+            iter=-1,
+            sweep=-1,
+            type='_time',
+            value=L.time + L.dt,
+        )
+
+
+class pySDC_integrator(TimeDiscretisation):
+    """
+    This class can be entered into Gusto as a time discretization scheme and will solve steps using pySDC.
+    It will construct a pySDC controller which can be used by itself and will be used within the time step when called
+    from Gusto. Access the controller via `pySDC_integrator.controller`. This class also has `pySDC_integrator.stats`,
+    which gathers all of the pySDC stats recorded in the hooks during every time step when used within Gusto.
+    """
+
+    def __init__(
+        self,
+        equation,
+        description,
+        controller_params,
+        domain,
+        field_name=None,
+        subcycling_options=None,
+        solver_parameters=None,
+        limiter=None,
```

Comment:
I think that not all of these options work from pySDC, e.g. the limiter... probably the sub cycling options aren't used and probably not the augmentation either... would it be better to remove them from the interface and docs?

Reply:
You're right! I removed those three parameters. I am not sure about the `options`. Seems to me like it might work when using Gusto for time integration because it adds a wrapper, but I don't actually know.